hyperspy/component.py

```diff
@@ -756,6 +756,9 @@ def summary(self):
                                                parameter.std,
                                                parameter.units)
 
+    def __tempcall2d__(self, p, x, y, onlyfree=True):
+        self.fetch_values_from_array(p, onlyfree=onlyfree)
+        return self.function(x, y)
```

Comment:
Actually the modification of the component would be somewhat problematic - all of the boundaries and other individual properties of a parameter would have to be rewritten to be able to use tuples. While it may be useful for support of even higher dimensions, I think it would drastically reduce readability and usability of models in general (of we would have to "unwrap" the tuples for users anyway, which sort of defeats the purpose).

I also had a look and it's possible to get rid of the coordinates argument for the `__tempcall__` altogether, however that makes it more complex (i.e. checking what kind of model the component is in), and the maintainability of the 1D model is reduced dramatically - namely convolved models now have part of the code in components, and it is not obvious at all that anything changes from convolved to non-convolved cases unless you actually look at the code.

Then again, maybe there can be more structural changes that make it clearer again, but I doubt it's worth it for now.

Besides, while 3d models do sound fancy and exciting, I think it makes little sense to support them yet - not only very few use-cases exist IMO, but the parameter space gets enormous with higher dimensionality, so likely completely new algorithms have to be employed. Even the 2D gaussian fit (which is/should be an extremely trivial example to test that everything works) given by Duncan takes surprisingly long on my machine.